src/modules/commander/commander.cpp

```diff
@@ -2216,8 +2217,11 @@ Commander::run()
 		}
 
 		/* RC input check */
-		if (!status_flags.rc_input_blocked && sp_man.timestamp != 0 &&
-		    (hrt_elapsed_time(&sp_man.timestamp) < (rc_loss_timeout * 1_s))) {
+		if (!status_flags.rc_input_blocked
+			&& (sp_man.timestamp > _last_sp_man.timestamp)
+			&& ((sp_man.timestamp - _last_sp_man.timestamp) < (rc_loss_timeout * 1_s))
+			&& (hrt_elapsed_time(&sp_man.timestamp) < (rc_loss_timeout * 1_s))
```

Comment:
hrt_elapsed_time is expensive CPU load wise and should be only used where absolutely required.

Reply:
+1

Reply:
I actually have timing numbers for a lot of these things if you're curious. An hrt_elapsed_time call takes a bit more than a microsecond, which for comparison is cheaper than most trig operations (single precision).